src/sage/rings/polynomial/multi_polynomial_ideal.py

```diff
@@ -1026,38 +1031,53 @@ def triangular_decomposition(self, algorithm=None, singular=singular_default):
 
         ALGORITHMS:
 
-        - ``singular:triangL`` - decomposition of self into triangular
+        - ``"singular:triangL"`` - decomposition of ``self`` into triangular
           systems (Lazard).
 
-        - ``singular:triangLfak`` - decomp. of self into tri.  systems
+        - ``"singular:triangLfak"`` - decomposition of ``self`` into triangular systems
           plus factorization.
 
-          - ``singular:triangM`` - decomposition of self into
-            triangular systems (Moeller).
+        - ``"singular:triangM"`` - decomposition of ``self`` into
+          triangular systems (Moeller).
 
         OUTPUT: a list `T` of lists `t` such that the variety of
         ``self`` is the union of the varieties of `t` in `L` and each
         `t` is in triangular form.
 
         EXAMPLES::
 
-            sage: P.<e,d,c,b,a> = PolynomialRing(QQ,5,order='lex')
+            sage: P.<e,d,c,b,a> = PolynomialRing(QQ, 5, order='lex'); P.rename("P")
```

Comment:
@mkoeppe I don't know what's going on here but this rename doesn't seem to be supported:
```
$ sage
┌────────────────────────────────────────────────────────────────────┐
│ SageMath version 10.0, Release Date: 2023-05-20                    │
│ Using Python 3.11.4. Type "help()" for help.                       │
└────────────────────────────────────────────────────────────────────┘
sage: P.<e,d,c,b,a> = PolynomialRing(QQ, 5, order='lex')
sage: P.rename("P")
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File /usr/lib/python3.11/site-packages/sage/structure/sage_object.pyx:121, in sage.structure.sage_object.SageObject.rename (build/cythonized/sage/structure/sage_object.c:2154)()
    120 try:
--> 121     self.__custom_name = str(x)
    122 except AttributeError:

AttributeError: 'sage.rings.polynomial.multi_polynomial_libsingular.MPolynomialRing_libsingular' object has no attribute '__custom_name'

During handling of the above exception, another exception occurred:

NotImplementedError                       Traceback (most recent call last)
Cell In[2], line 1
----> 1 P.rename("P")

File /usr/lib/python3.11/site-packages/sage/structure/sage_object.pyx:123, in sage.structure.sage_object.SageObject.rename (build/cythonized/sage/structure/sage_object.c:2203)()
    121             self.__custom_name = str(x)
    122         except AttributeError:
--> 123             raise NotImplementedError("object does not support renaming: %s" % self)
    124 
    125 def reset_name(self):

NotImplementedError: object does not support renaming: Multivariate Polynomial Ring in e, d, c, b, a over Rational Field
sage: 
```

It's not working for me in 10.1.beta7 either. I don't know what is going on, since I had not seen this error before and this was merged in 10.1.beta1.

Reply:
Fixed in #35749 already

Reply:
> Fixed in #35749 already

Is that the right PR? That one is 157 commits and mentions nothing of the issue.

Reply:
See the comment

https://github.com/sagemath/sage/pull/35749/files#r1257472203

Reply:
From the top of the description of #35749:
- Fixing the handling of file-level # optional tags.
- Some files were not being doctested; fixing the recently introduced errors in doctests.

Reply:
Ok, I thought I was getting crazy. I was trying to test #35977 and getting this error I never saw before but now I could reproduce with old branches.

Now there are two more regressions in singular 4.3.2p3.